### PR TITLE
Add a leading underscore to internal and private functions.

### LINKED ISSUE #1176: Add a leading underscore to internal functions
It would be good to add an indication to function names when they are private. Let's add an _ in front of the name.

### PR DESCRIPTION
**Pull request by elopio**
Fixes #1176 

# 🚀 Description

Add a leading underscore for private and internal functions.

- [x] 📘 I've reviewed the [OpenZeppelin Contributor Guidelines](../blob/master/CONTRIBUTING.md)
- [x] ✅ I've added tests where applicable to test my new functionality.
- [x] 📖 I've made sure that my contracts are well-documented.
- [x] 🎨 I've run the JS/Solidity linters and fixed any issues (`npm run lint:all:fix`).

### CODE REVIEW

**Comment by elopio** [comment]
I must say I don't like this for internal functions. They are part of the public API of a contract, and I think the underscore makes this less clear.
However, I think the problem is bigger than this. We have already discussed that we rely too much on inheritance and that makes everything more complicated. I'm ok delaying the bulk of this discussion for 3.0, after we experiment with better encapsulation, more private things and composition.

File: contracts/mocks/ERC721TokenMock.sol [hunk 1]
```diff
@@ -29,7 +29,7 @@ contract ERC721TokenMock is ERC721Token {
     super._setTokenURI(_tokenId, _uri);
   }
 
-  function _removeTokenFrom(address _from, uint256 _tokenId) public {
-    super.removeTokenFrom(_from, _tokenId);
+  function removeTokenFrom(address _from, uint256 _tokenId) public {
+    super._removeTokenFrom(_from, _tokenId);
```
Comment (nventuro) [inline contracts/mocks/ERC721TokenMock.sol]: I think `super.` is not needed here

Comment (elopio) [reply contracts/mocks/ERC721TokenMock.sol]: supers removed. This is ready for a new review.

**Comment by elopio** [comment]
@nventuro it is possible to have solium rules for almost everything! :)
But let's push them to next week. Once there are no more 2.0 issues, I can merge all the PRs in progress in our solium plugin, write the missing ones, make a release and run it on this project to see if we missed something.

**Comment by nventuro** [comment]
For some reason the diff is huge, including the changes from #1252. @elopio can you try merging master to see if that fixes it?

**Pull request merged**